ariadne/cardano/test/backend/Test/Spec/Wallets.hs

```diff
@@ -128,3 +134,73 @@ spec = describe "CreateWallet" $ do
                                  let wid = WalletIdHdSeq (hdRoot ^. hdRootId)
                                  mbEsk <- Keystore.lookup wid (wallet ^. Internal.walletKeystore)
                                  mbEsk `shouldSatisfy` isJust
+
+    describe "Update wallet passsword" $ do
+        prop "works as expected in the happy path scenario" $ withMaxSuccess 50 $ do
+            monadicIO $ do
+                currentPassword <- genSpendingPassword
+                newPassword <- genSpendingPassword
+                request <- genNewWalletRq currentPassword
+
+                pm <- pick arbitrary
+                withLayerInMemoryStorage pm $ \layer wallet -> do
+                    liftIO $ do
```

Comment:
Perhaps it makes sense to put `liftIO` one line above, leave only one `do` and save some indentation space.